src/pykeen/evaluation/expectation.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+"""Utilities for calculating the expected value of metrics."""
+
+from typing import Sequence, Union
+
+import numpy as np
+
+from .metrics import ALL_TYPE_FUNCS
+
+__all__ = [
+    "numeric_expected_value",
+    "expected_mean_rank",
+    "expected_hits_at_k",
+]
+
+
+def numeric_expected_value(
+    metric: str,
+    num_candidates: Union[Sequence[int], np.ndarray],
+    num_samples: int,
+) -> float:
+    """
+    Compute expected metric value by summation.
+
+    Depending on the metric, the estimate may not be very accurate and converage slowly, cf.
+    https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.rv_discrete.expect.html
+    """
+    metric_func = ALL_TYPE_FUNCS[metric]
+    num_candidates = np.asarray(num_candidates)
+    generator = np.random.default_rng()
+    expectation = 0
+    for _ in range(num_samples):
+        ranks = generator.integers(low=0, high=num_candidates)
+        expectation += metric_func(ranks)
+    return expectation / num_samples
+
+
+def expected_mean_rank(
+    num_candidates: Union[Sequence[int], np.ndarray],
+) -> float:
+    r"""
+    Calculate the expected mean rank under random ordering.
+
+    .. math ::
+
+        E[MR] = \frac{1}{n} \sum \limits_{i=1}^{n} \frac{1 + CSS[i]}{2}
+              = \frac{1}{2}(1 + \frac{1}{n} \sum \limits_{i=1}^{n} CSS[i])
+
+    :param num_candidates:
+        the number of candidates for each individual rank computation
+
+    :return:
+        the expected mean rank
+    """
+    return 0.5 * (1 + np.mean(np.asanyarray(num_candidates)))
```

Comment:
we could also make this a function of the respective metric class

Reply:
Yes! Getting there, one tiny PR at a time :)